forem/forem
# API v1 without api key restriction

## What type of PR is this? (check all applicable)


## Description

After reflecting and discussing internally we're rolling back the decision to restrict CORS + unauthenticated endpoints from the v1 API.

This PR tackles the rollback of that logic without removing the feature flag. Meanwhile #18213 will only include the changes to remove the feature flag (separate PR).

## Related Tickets & Documents

- Related #18213

## QA Instructions, Screenshots, Recordings

1. Enable `:api_v1` Feature Flag
1. Create an API Key in /settings/extensions
1. Send in a PUT request to `/api/users/:id/unpublish` with the API Key and `application/vnd.forem.api-v1+json` in the Accept header
1. Make API requests to the local server
   - The API is exactly the same as v0 (specs are unchanged there) but it should include a few more endpoints regarding unpublishing posts + suspending a user

### UI accessibility concerns?

N/A

## Added/updated tests?


## [Forem core team only] How will this change be communicated?


## [optional] What gif best describes this PR or how it makes you feel?

![](https://media.giphy.com/media/9Dnxbpdm15gRRslPRz/giphy.gif)
